larq_zoo/quicknet.py

```diff
@@ -0,0 +1,170 @@
+from typing import Optional, Sequence
+
+import larq as lq
+import tensorflow as tf
+from zookeeper import Field, factory
+
+from larq_zoo import utils
+from larq_zoo.model_factory import ModelFactory
+
+
+@factory
+class QuickNetFactory(ModelFactory):
+
+    num_layers: int = Field(15)
+    initial_filters: int = Field(64)
+
+    input_quantizer = Field(lambda: lq.quantizers.SteSign(clip_value=1.25))
+    kernel_quantizer = Field(lambda: lq.quantizers.SteSign(clip_value=1.25))
+    kernel_constraint = Field(lambda: lq.constraints.WeightClip(clip_value=1.25))
+
+    @property
+    def spec(self):
+        spec = {
+            15: ([2, 3, 4, 4], [64, 128, 256, 512]),
+        }
+        try:
+            return spec[self.num_layers]
+        except Exception:
+            raise ValueError(f"Only specs for layers {list(self.spec.keys())} defined.")
+
+    def LCEFirstLayer(self, x: tf.Tensor) -> tf.Tensor:
+        x = tf.keras.layers.Conv2D(
+            int(self.initial_filters / 8),
+            (3, 3),
+            strides=2,
+            kernel_initializer="he_normal",
+            padding="same",
+            use_bias=False,
+        )(x)
+        x = tf.keras.layers.DepthwiseConv2D(
+            (3, 3),
+            depth_multiplier=8,
+            strides=2,
+            kernel_initializer="he_normal",
+            padding="same",
+            use_bias=False,
+            activation="relu",
+        )(x)
+
+        x = tf.keras.layers.BatchNormalization(momentum=0.9, epsilon=1e-5)(x)
+        return x
```

Comment:
Probably can just `return tf.keras. ...`

Reply:
I'll update this now